session/txn.go

```diff
@@ -598,7 +598,7 @@ func (txn *LazyTxn) Wait(ctx context.Context, sctx sessionctx.Context) (kv.Trans
 		}
 		txn.lazyUniquenessCheckEnabled = !sctx.GetSessionVars().ConstraintCheckInPlacePessimistic
 		// set resource group name for kv request such as lock pessimistic keys.
-		txn.SetOption(kv.ResourceGroupName, sctx.GetSessionVars().ResourceGroupName)
+		kv.SetTxnResourceGroup(txn, sctx.GetSessionVars().ResourceGroupName)
```

Comment:
Can this be put into `txn.changePendingToValid` (though it's only called in this Wait function), to avoid this being forgotten if there's going to be more invocations to `changePendingToValid` in the future?

Reply:
done.